### PR TITLE
Planner: Don't force ascent

### PR DESCRIPTION
**Pull request by atdotde**
The planner should not always ascent from the depth of
the last manually entered waypoint but consider the
possibility that it should first top where we are
before the next stop depth has cleared.

Reported-by: David Carron

Signed-off-by: Robert C. Helling <helling@atdotde.de>

### Describe the pull request:

- [x] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### CODE REVIEW

File: core/planner.c [hunk 1]
```diff
@@ -757,7 +757,7 @@ bool plan(struct deco_state *ds, struct diveplan *diveplan, struct dive *dive, i
 
 	/* Find the first potential decostopdepth above current depth */
 	for (stopidx = 0; stopidx < decostoplevelcount; stopidx++)
-		if (*(decostoplevels + stopidx) >= depth)
```
Comment (bstoeger) [inline core/planner.c:760]: I can't comment on the patch itself. Just a coding-style comment:
Could we change the condition to `decostoplevels[stopidx] > depth`?
As it stands, that is super-cryptic.

Comment (atdotde) [reply core/planner.c:760]: You are right. That code is ancient and was written before my time...

**Comment by dirkhh** [comment]
The commit contains a libdivecomputer change.
NAK

**Comment by atdotde** [comment]
Argh, this happens when I rebase...

**Review by dirkhh — APPROVED** [review]
thanks for fixing it

**Pull request merged**